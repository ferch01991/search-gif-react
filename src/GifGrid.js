import React from 'react'
import { useFetchGifs } from './hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
// import { getGifs } from './helpers/getGifs'

export const GifGrid = ({ category }) => {
    
    // data: images => es el nombre que se le quiere dar
    const { data:images, loading } = useFetchGifs(category);

    // const [images, setImages] = useState([])

    // useEffect(() => {
    //     // Se ejecuta esta instruccion cuando el componente es renderizado pro primera vez
    //     getGifs(category) // funcion del helpers
    //         .then(setImages)
    //         // .then( (imgs) => setImages(imgs))
    //     // Si la category cambia se vuelve a ejecutar useEffect
    //     }, [category] /* Arreglo de dependencias*/)



    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> {category} </h3>

            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            
            <div className="card-grid">

                {
                    images.map(img => (
                        // Los parentesis hacen qeu sea un return implicito
                        <GifGridItem
                            key={img.id}
                            // spredth
                            {...img}
                        />
                    ))
                }
            </div>
        </>

    )
}
