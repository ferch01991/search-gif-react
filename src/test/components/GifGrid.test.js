import { GifGrid } from '../../GifGrid'
import React from 'react'

import { shallow } from "enzyme"
import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('pruebas en el componente <GifGrid />', () => {

    const category = 'One Punch' 

    test('should snapshop de GifGrid', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
            
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })

    // useFetchGifs regresa la informacion
    test('debe mostrar items cuando se carga imagenes useFetchGifs ', () => {
        
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost.com',
                title: 'Cualquier cosa'
            },
            {
                id: '4754',
                url: 'https://localhost.com',
                title: 'Cualquier cosa'
            }
        ]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        // En principio no deberia haber parrafos
        expect(wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length)

    })
    
    
})
