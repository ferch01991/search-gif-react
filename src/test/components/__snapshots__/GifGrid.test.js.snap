// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GifGrid /> debe mostrar items cuando se carga imagenes useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost.com"
    />
    <GifGridItem
      id="4754"
      key="4754"
      title="Cualquier cosa"
      url="https://localhost.com"
    />
  </div>
</Fragment>
`;

exports[`pruebas en el componente <GifGrid /> should snapshop de GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
